test/vfd_swmr_bigset_writer.c

```diff
@@ -218,56 +251,74 @@ usage(const char *progname)
             "-c cols:	       `cols` columns per chunk\n"
             "-d 1|one|2|two|both:  select dataset expansion in one or\n"
             "                      both dimensions\n"
+            "-l tick_num:          expected maximal number of ticks from\n"
+            "                      the writer's finishing creation to the reader's finishing validation\n"
+            "-N:                   do not use named pipes\n"
             "-n iterations:        how many times to expand each dataset\n"
             "-r rows:	       `rows` rows per chunk\n"
             "-s datasets:          number of datasets to create\n"
-            "-u ms:                milliseconds interval between updates\n"
-            "                      to %s.h5\n"
             "\n",
-            progname, progname);
+            progname);
     exit(EXIT_FAILURE);
 }
 
-static void
+static bool
 make_quadrant_dataspace(state_t *s, quadrant_t *q)
 {
-    q->space = H5Screate_simple(NELMTS(s->chunk_dims), s->chunk_dims,
-                                s->two_dee ? two_dee_max_dims : s->one_dee_max_dims);
+    if ((q->space = H5Screate_simple(NELMTS(s->chunk_dims), s->chunk_dims,
+                                     s->two_dee ? two_dee_max_dims : s->one_dee_max_dims)) < 0) {
+        fprintf(stderr, "H5Screate_simple failed\n");
+        TEST_ERROR;
+    }
+
+    if (H5Sselect_hyperslab(q->space, H5S_SELECT_SET, q->start, q->stride, q->count, q->block) < 0) {
+        fprintf(stderr, "H5Sselect_hyperslab failed\n");
+        TEST_ERROR;
+    }
 
-    if (q->space < 0) {
-        errx(EXIT_FAILURE, "%s.%d: H5Screate_simple failed", __func__, __LINE__);
+    return true;
+
+error:
+    H5E_BEGIN_TRY
+    {
+        H5Sclose(q->space);
     }
+    H5E_END_TRY;
 
-    if (H5Sselect_hyperslab(q->space, H5S_SELECT_SET, q->start, q->stride, q->count, q->block) < 0)
-        errx(EXIT_FAILURE, "%s: H5Sselect_hyperslab failed", __func__);
+    return false;
 }
 
-static void
+static bool
 state_init(state_t *s, int argc, char **argv)
 {
     unsigned long     tmp;
     int               ch;
     unsigned          i;
-    const hsize_t     dims = 1;
-    char              tfile[PATH_MAX];
+    const hsize_t     dims  = 1;
+    char *            tfile = NULL;
     char *            end;
-    unsigned long     millis;
     quadrant_t *const ul = &s->quadrants.ul, *const ur = &s->quadrants.ur, *const bl = &s->quadrants.bl,
                       *const br = &s->quadrants.br, *const src = &s->quadrants.src;
     const char *personality;
 
     *s = state_initializer();
-    esnprintf(tfile, sizeof(tfile), "%s", argv[0]);
-    esnprintf(s->progname, sizeof(s->progname), "%s", basename(tfile));
 
-    while ((ch = getopt(argc, argv, "FMSVWa:bc:d:n:qr:s:u:")) != -1) {
+    if (H5_basename(argv[0], &tfile) < 0) {
+        fprintf(stderr, "H5_basename failed\n");
+        TEST_ERROR;
+    }
+
+    esnprintf(s->progname, sizeof(s->progname), "%s", tfile);
+
+    if (tfile)
```

Comment:
The standard says that freeing a NULL pointer is acceptable so you don't need to check for null before calling free().

Reply:
Changed.